client/allocator/src/freeing_bump.rs

```diff
@@ -913,24 +1007,24 @@ mod tests {
 
 	#[test]
 	fn deallocate_needs_to_maintain_linked_list() {
-		let mut mem = [0u8; 8 * 2 * 4 + ALIGNMENT as usize];
```

Comment:
Hmm.... since `Memory` (which is the public interface) can only return `pages()`, and we only support growing the memory in pages these tests are kind-of nonsense now. You create a `MemoryInstance` here with a size of 72, but when `pages()` is called it will return `1` and when `size()` is called it will return `65536`, neither of which is technically true, and should never actually happen in production code. Functionally for the tests I guess it's fine, but from a conceptual and documentation standpoint it doesn't really make much sense. Would be nice to enforce that with appropriate `assert!`s and adjust the tests to reflect that reality. (And maybe change `with_size` to `with_pages`?)

Reply:
We are not testing here the public interface of `Memory`. We are testing here some specific internals that should ensure that we free up space accordingly. While you are right that this will not happen in reality (which was already true before this pr), we are just testing here some specific part of the implementation. I think we should keep it.